File: problem4_matlab/problem4.m
%% Refresh
%# Created by:
%##################################
%#########SOHAM SARKAR#############
%#####Arizona State University#####
%##################################

% # Do NOT distribute without written permission from Soham Sarkar
% # Do NOT use it for any commercial purpose
% 
% # Contact email: ssarka30@asu.edu
% # Last update: May 14, 2024
clc
clear all
close all
%% Main Body of Code
disp("===================================================")
disp("Enter any two numbers")
a = input("Enter the first number ");
b = input("Enter the second number ");

fprintf('First Element: %d \n',car(cons(a,b)))
fprintf('Last Element: %d \n',cdr(cons(a,b)))
disp("===================================================")

function [pair] = cons(a,b)
    pair = [a,b];
    return
end

function elem1 = car(pair)
    elem1 = pair(1);
    return
end

function elem2 = cdr(pair)
    elem2 = pair(end);
    return
end

File: problem4_python/problem4.py
# -*- coding: utf-8 -*-
"""
Created on Tue May 14 03:09:30 2024

@author: soham
"""

print("Enter any two numbers: \n")

a = input('Enter the first number \n')
b = input('Enter the second number \n')

class cons:
    def __init__(self,a,b):
        self.a = a
        self.b = b
        self.f = [self.a, self.b]
        

pair = cons(a,b)
    
def car(pair):
    return pair.a

def cdr(pair):
    return pair.b


# run in terminal
# cdr(pair)

# car(pair)


output1= cdr(cons(1,2))

output2= car(cons(1,2))

print('cdr output',output1)

print('car output',output2)

